scripts/ci/publish

```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+: ${SNAP_CI?This script is designed to be run within SnapCI. Aborting in a cowardly fashion.}
+
+
+# Login to NPM
+local npmrc_contents="//registry.npmjs.org/:_authToken=${NPM_TOKEN}"
```

Comment:
Found the details. 

I was intermittently getting "npm: command not found" in our snap builds - e.g. https://snap-ci.com/meetearnest/autopublish/branch/master/logs/defaultPipeline/14/test

Here's what I eventually got from Snap support: 

> Alright, we think we've figured out the issue and have a workaround for you! :) 
> 
> So when you select a version of node from the drop-down menu, it runs in the background before your build commands even get started. When you run the build for the first time on a brand new container, NVM is able to install the version of node and save the configuration. When we have your previous run cached, NVM fails to save the config because of a bug with variable expansion. (more on that here, if you're interested: https://github.com/npm/npm/issues/10782) 
> 
> The workaround is to create a .npmrc file, which makes this a non-issue. For now we recommend that you create a secure file for .npmrc in your home directory: /var/go/ which can be empty, just as long as it exists. 

I followed their suggestion and didn't see the issue any more.